compiler/src/iree/compiler/Codegen/Utils/VectorOpUtils.cpp

```diff
@@ -63,15 +41,9 @@ VectorContractOpInfo::inferOpKind(MLIRContext *ctx,
     outNDims.push_back(*maps[2].getResultPosition(getAffineDimExpr(n, ctx)));
   }
 
-  if (outM < outN) {
-    if (lhsM < lhsKDim) {
-      if (rhsN < rhsKDim) {
-        return OpKind::MK_NK_MN;
-      }
-      return OpKind::MK_KN_MN;
-    }
-  }
-  return OpKind::UNKNOWN;
+  int64_t k = contractionDims.k.back();
+  lhsKDim = *maps[0].getResultPosition(getAffineDimExpr(k, ctx));
```

Comment:
Noice! interesting that we did that for m and n dim before but not for K dim haha

Reply:
yeah...